src/backend/src/utils/transporter.utils.ts

```
@@ -0,0 +1,63 @@
+import nodemailer from 'nodemailer';
+import { google } from 'googleapis';
+import SMTPTransport from 'nodemailer/lib/smtp-transport';
+
+const { OAuth2 } = google.auth;
+
+const createTransporter = async () => {
+  try {
+    const oauth2Client = new OAuth2(
+      process.env.CLIENT_ID,
+      process.env.CLIENT_SECRET,
+      'https://developers.google.com/oauthplayground'
+    );
+
+    oauth2Client.setCredentials({
+      refresh_token: process.env.REFRESH_TOKEN
+    });
+
+    let accessToken: string | null | undefined;
+
+    await oauth2Client.getAccessToken((err, token) => {
+      if (err) {
+        console.log('*ERR: ', err);
+        throw err;
+      }
+      accessToken = token;
+    });
+
+    const transporter = nodemailer.createTransport({
+      service: 'gmail',
+      auth: {
+        type: 'OAuth2',
+        user: process.env.USER_EMAIL,
+        accessToken: accessToken?.toString(),
+        clientId: process.env.CLIENT_ID,
+        clientSecret: process.env.CLIENT_SECRET,
+        refreshToken: process.env.REFRESH_TOKEN
+      }
+    });
+    return transporter;
+  } catch (err) {
+    return err;
+  }
+};
+
+const sendMailToAdvisor = async (subject: string, text: string) => {
+  try {
+    //this sends an email from our email to our advisor: professor Goldstone
+    const mailOptions = {
+      from: process.env.USER_EMAIL,
+      to: 'mckee.p@northeastern.edu',
+      subject,
+      text
+    };
+
+    const emailTransporter = (await createTransporter()) as nodemailer.Transporter<SMTPTransport.SentMessageInfo>;
+    await emailTransporter.sendMail(mailOptions);
+  } catch (err) {
+    console.log('ERROR: ', err);
```

this should throw an exception if it fails